### PR TITLE
OWA-35: Implement functionality to upload an OWA

### PR DESCRIPTION
**Pull request by jomadoye**
# JIRA TICKET NAME:
OWA-35: [Implement functionality to upload an OWA](https://issues.openmrs.org/browse/OWA-35)

## SUMMARY:
This PR Implement functionality to upload an OWA

### CODE REVIEW

**Comment by paulupendo** [comment]
@andela-jomadoye Great job Jed. I have gone through the implementation to upload an OWA and I have not seen use of the endpoint 'ws/rest/owa/addapp' does this mean we won't need it during the upload and if so how does the uploaded app integrate with the backend?

File: app/js/components/homePage/manageApps/AddAddon.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export default function AddAddon({ handleClear }) {
+  return (
+    <div className="container-fluid">
+      <fieldset className="scheduler-border">
+        <legend className="scheduler-border">Upload Addon Package:</legend>
+        <div className="col-sm-9">
+          <input 
+            type="file" 
+            accept=".zip,.omod"
```
Comment (paulupendo) [inline app/js/components/homePage/manageApps/AddAddon.jsx]: Are we ready to handle .omod files with the app?

**Comment by jomadoye** [comment]
@paulupendo I'm still working on this PR that's why I haven't requested code-review. 👍

File: app/js/components/homePage/manageApps/AddAddon.jsx (hunk 1 above)
Comment (jomadoye) [reply app/js/components/homePage/manageApps/AddAddon.jsx]: Not yet.

**Comment by jomadoye** [comment]
This PR is ready for review, thanks @paulupendo for helping out with the blocker.

File: app/js/components/homePage/manageApps/manageApps.jsx [hunk 2]
```diff
@@ -7,22 +7,57 @@
...
+
+  handleUpload() {
+    const applicationDistribution = location.href.split('/')[2];
+    const url = location.href.split('/')[3];
+    let apiBaseUrl = `/${applicationDistribution}/${url}/ws/rest`;
+    this.requestUrl ='/owa/addapp';
+
+    const fd = new FormData();
+    fd.append('file', document.getElementById('fileInput').files[0]);
+    $.ajax({
```
Comment (paulupendo) [inline app/js/components/homePage/manageApps/manageApps.jsx]: @andela-jomadoye You probably will have to declare $ as a global variable and define it to require query, else tests will fail 😄

File: app/js/components/homePage/manageApps/manageApps.jsx [hunk 3]
```diff
@@ -7,22 +7,57 @@
...
+    $(":file").filestyle({btnClass: "btn-primary"});
+  }
+
+  handleUpload() {
+    const applicationDistribution = location.href.split('/')[2];
+    const url = location.href.split('/')[3];
+    let apiBaseUrl = `/${applicationDistribution}/${url}/ws/rest`;
+    this.requestUrl ='/owa/addapp';
+
+    const fd = new FormData();
```
Comment (paulupendo) [inline app/js/components/homePage/manageApps/manageApps.jsx]: Renaming the constant fd to maybe data would be great.

**Review by paulupendo — COMMENTED** [review]
@andela-jomadoye You are a great team mate giving quality outputs.

File: app/js/components/homePage/manageApps/manageApps.jsx (hunk 2 above)
Comment (jomadoye) [reply app/js/components/homePage/manageApps/manageApps.jsx]: @paulupendo Test is passing 😄.

**Pull request merged**